contents/code/tilelist.js

```diff
@@ -187,6 +187,12 @@ TileList.prototype.connectSignals = function(client) {
             tile.onClientDesktopChanged(client);
         }
     });
+    client.screenChanged.connect(function() {
```

Comment:
That easy?

My main concern, and I've brought this up with the KWin folks before, is ordering of the signals.

Does screenChanged always fire before e.g. moveResized?

Reply:
`screenChanged` fires when moving finishes, just before `clientFinishUserMovedResized`. I was also concerned about the ordering. I did all kinds of things to my windows - moving between screens using window menu, dragging the whole windows, even resizing over the edge of the screen and so far everything seems to work just fine.

Reply:
Okay, so let's merge it and see what happens!